Classes/Auth/TMAuthenticationResponseProcessor.m

```diff
@@ -72,8 +74,14 @@ - (TMURLSessionRequestCompletionHandler)sessionCompletionBlock {
                         statusCode = passwordStatusCode;
                     }
 
-                    
-                    self.callback(nil, errorWithStatusCode(statusCode));
+                    TMResponseParser *responseParser = [[TMResponseParser alloc] initWithData:data
+                                                                                  URLResponse:response
+                                                                                        error:nil
+                                                                                serializeJSON:YES];
+
+                    NSDictionary *fullResponseJSON = [responseParser fullJSON];
```

Comment:
You shouldn't need to do this - use the the `parse` method and grab it off there?